File: chapter2/chapter2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"

class PersonNode : public rclcpp::Node
{
private:
    // 声明
    std::string name_;
    int age_;

public:
    PersonNode(const std::string &node_name, const std::string &name, const int &age)
    :Node(node_name) // 调用父类的构造函数，等同于python中的super().__init__()
    {
        this->name_ = name;
        this->age_ = age;
    }

    void eat(const std::string &food_name)
    {
        RCLCPP_INFO(this->get_logger(),"我是%s, %d岁，爱吃%s", this->name_.c_str(), this->age_, food_name.c_str());
    }
};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("lisi", "李四", 18);
    RCLCPP_INFO(node->get_logger(), "hello cpp node!");
    node->eat("鱼香肉丝");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: chapter2/chapter2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self, node_name: str, name_value: str, age_value: int) -> None:
        print('PersonNode __init__方法被调用了，添加了两个属性')
        super().__init__(node_name)
        self.name = name_value
        self.age = age_value

    def eat(self, food_name: str):
        """
        方法：吃东西
        :food_name 食物名字
        """
        # print(f"{self.name}, {self.age}岁，爱吃{food_name}")
        self.get_logger().info(f"{self.name}, {self.age}岁，爱吃{food_name}")


def main():
    rclpy.init()
    node = PersonNode('zhangsan', '法外狂徒张三', 18)
    node.eat('鱼香肉丝')
    rclpy.spin(node)
    rclpy.shutdown()